src/sonic-build-hooks/scripts/buildinfo_base.sh

```diff
@@ -242,10 +243,24 @@ download_packages()
         return $result
     fi
 
-    $REAL_COMMAND "${parameters[@]}"
-    result=$?
+    # Retry if something super-weird has happened. Use --retry-connrefused
+    # option for wget/curl.
+    # --retry-connrefused - Consider "connection refused" a transient error and
+    # try again. Normally wget/curl gives up on a URL when it is unable to
+    # connect to the site because failure to connect is taken as a sign that the
+    # server is not running at all and that retries would not help. This option
+    # is for mirroring unreliable sites whose servers tend to disappear for
+    # short periods of time.
+    for ((i = 1; i <= GET_RETRY_COUNT; i++)); do
+        $REAL_COMMAND --retry-connrefused "${parameters[@]}"
```

Comment:
>--retry-connrefused [](http://example.com/codeflow?start=22&length=19)

REAL_COMMAND may be commands other than `curl`. So the paramter `--retry-connrefused` may be not applicable. If you need `curl` to retry even refused, change the file `src/sonic-build-hooks/hooks/curl`. And you can also increase `--retry` parameter there.

Reply:
In the comment above I mentioned that this option is applicable to both `curl` and `wget`. It works for both. If in the future someone will try to add another method (so unlikely, cause curl and wget are fast and designed to run in shell) it will fail with the error during the build, so no hidden errors here. I think we can safely leave it as it is.

Reply:
@xumia Please help review. What are the possible options for REAL_COMMAND? Is it possible to be `pip3`, `apt-get`, etc?

Reply:
> `download_package` also uses curl to calculate md5sum of downloaded file (check function `get_url_version`).
> So these additional retries for wget/curl are useless if `get_url_version` failed to download a file.
> Also this double downloading (wget+curl or curl+curl) is not good for poor connections.

@k-v1, the function you mentioned uses the original `curl`, so it doesn't correlate with the code I've added.

Reply:
> @xumia Please help review. What are the possible options for REAL_COMMAND? Is it possible to be `pip3`, `apt-get`, etc?

@xumia gentle ping here

Reply:
If this option is for wget/curl, then you can add it into
src/sonic-build-hooks/hooks/curl
src/sonic-build-hooks/hooks/wget

Adding here is too general.

Reply:
Sure, I will move it there and update PR

Reply:
Done